src/Domain/BlockLogicalCoordinates.cpp

```diff
@@ -38,11 +38,26 @@ std::vector<block_logical_coord_holder<Dim>> block_logical_coordinates(
     // case, choose the first matching block (and this block will have
     // the smallest block_id).
     for (const auto& block : domain.blocks()) {
-      const auto inv = block.coordinate_map().inverse(x_frame);
-      if(inv) {
-        x_logical = inv.get();
+      if (block.is_time_dependent()) {
+        const auto moving_inv =
+            block.moving_mesh_grid_to_inertial_map().inverse(x_frame);
```

Comment:
Does this loop continue looping after it finds a point? Can you just `break;` if it's found?

Reply:
I think that's handled on line new 74/old 58 below if I understand what you're saying correctly

Reply:
ah, good, thanks -- I miscounted the braces.